pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -347,6 +332,52 @@ type fakeChainReader struct {
 	lock        sync.Mutex
 }
 
+type fakeChainWriter struct {
+	types.ChainWriter
+	cr *fakeChainReader
+}
+
+func (f *fakeChainWriter) SubmitTransaction(ctx context.Context, contractName, method string, args any, transactionID string, toAddress string, meta *types.TxMeta, value *big.Int) error {
+	switch method {
+	case MethodSettingStruct:
+		v, ok := args.(TestStruct)
+		if !ok {
+			return fmt.Errorf("unexpected type %T", args)
+		}
+		f.cr.SetTestStructLatestValue(&v)
+	case MethodSettingUint64:
+		v, ok := args.(PrimitiveArgs)
+		if !ok {
+			return fmt.Errorf("unexpected type %T", args)
+		}
+		f.cr.SetUintLatestValue(v.Value, ExpectedGetLatestValueArgs{})
+	case MethodTriggeringEvent:
+		v, ok := args.(TestStruct)
+		if !ok {
+			return fmt.Errorf("unexpected type %T", args)
+		}
+		f.cr.SetTrigger(&v)
+	case "batchChainWrite":
```

Comment:
shouldn't this be a const as well?

Reply:
Addressed that in [this comment](https://github.com/smartcontractkit/chainlink-common/pull/664#discussion_r1742110028)